stan/math/prim/err/check_consistent_sizes_mvt.hpp

```diff
@@ -3,31 +3,83 @@
 
 #include <stan/math/prim/err/check_consistent_size_mvt.hpp>
 #include <stan/math/prim/fun/size_mvt.hpp>
+#include <stan/math/prim/meta/require_generics.hpp>
 #include <algorithm>
 
 namespace stan {
 namespace math {
+namespace internal {
+
+/** Base case of recursion, this function is a no-op. */
+inline void check_consistent_with_size_mvt(const char*, size_t) { return; }
+
+/**
+ * Check that the provided inputs are either non-vectors, or vectors of the
```

Comment:
This doc is identical to that for `check_consist_with_size`.  This doc shold explain what the `mvt` means in the suffix.

Reply:
Whoops. Meant to fix that. Thanks!

Reply:
I'm also cleaning up check_consistent_size_mvt (Note: "size" not "sizes"). The logic and documentation there is pretty confusing and can be simplified.